plugins/outputs/kinesis/kinesis.go

```diff
@@ -154,26 +155,33 @@ func (k *KinesisOutput) SetSerializer(serializer serializers.Serializer) {
 	k.serializer = serializer
 }
 
-func writekinesis(k *KinesisOutput, r []*kinesis.PutRecordsRequestEntry) time.Duration {
+func (k *KinesisOutput) mockKinesisService(svc kinesisiface.KinesisAPI) {
+	k.svc = svc
+}
+
+func (k *KinesisOutput) writeKinesis(r []*kinesis.PutRecordsRequestEntry) time.Duration {
+
 	start := time.Now()
 	payload := &kinesis.PutRecordsInput{
 		Records:    r,
 		StreamName: aws.String(k.StreamName),
 	}
 
+	resp, err := k.svc.PutRecords(payload)
+	if err != nil {
+		log.Printf("E! kinesis: Unable to write to Kinesis : %s", err.Error())
+		return time.Since(start)
+	}
+
 	if k.Debug {
-		resp, err := k.svc.PutRecords(payload)
-		if err != nil {
-			log.Printf("E! kinesis: Unable to write to Kinesis : %s", err.Error())
-		}
 		log.Printf("I! Wrote: '%+v'", resp)
+	}
 
-	} else {
-		_, err := k.svc.PutRecords(payload)
-		if err != nil {
-			log.Printf("E! kinesis: Unable to write to Kinesis : %s", err.Error())
-		}
+	var failed = *resp.FailedRecordCount
+	if failed > 0 {
+		log.Printf("E! kinesis: Unable to write %+v of %+v record(s) to Kinesis", failed, len(r))
```

Comment:
Previously we have been silently dropping metrics

Reply:
would be great if someone updated this plugin to use 
```go
  Log telegraf.Logger `toml:"-"`
```
then the code could use `k.Log.Debug()` without conditionals, etc.
tests can set the logger with `testutil.Logger{}`

Reply:
I plan to do a few more PRs after this one.  Be glad to tackle that for you.